pkg/util/provider/machinecontroller/controller.go

```diff
@@ -51,8 +51,14 @@ const (
 	ClassAnnotation = "machine.sapcloud.io/class"
 	// MachineIDAnnotation is the annotation used to identify a machine ID
 	MachineIDAnnotation = "machine.sapcloud.io/id"
-	// DeleteFinalizerName is the finalizer used to identify the controller acting on an object
-	DeleteFinalizerName = "machine.sapcloud.io/machine-controller-manager"
+	// MCMFinalizerName is the finalizer used to tag dependecies before deletion
+	// of the object. This finalizer is carried over from then MCM
+	MCMFinalizerName = "machine.sapcloud.io/machine-controller-manager"
+	// MCFinalizerName is the new finalizer created for the external
+	// machine controller to seperate it from the older MCMFinalizerName
+	// This is useful add finalizers to secrets where in-tree controller might
+	// still be looking for mappings and acting on it.
+	MCFinalizerName = "machine.sapcloud.io/machine-controller"
```

Comment:
```suggestion
	// of the object. This finalizer is carried over from the MCM
	MCMFinalizerName = "machine.sapcloud.io/machine-controller-manager"
	// MCFinalizerName is the finalizer created for the external
	// machine controller to differentiate it from the MCMFinalizerName
	// This finalizer is added only on secret-objects to avoid race between in-tree and out-of-tree controllers.
	// This is a stopgap solution to resolve: https://github.com/gardener/machine-controller-manager/issues/486. 
	MCFinalizerName = "machine.sapcloud.io/machine-controller"
```

Reply:
Please accept the suggestion only we agree on the overall review.